tests/mocked_plans/sample_plans/base.yaml

```diff
@@ -4,6 +4,9 @@ plan: |
         Verify that all OpenStack applications are in idle state
         Back up MySQL databases
         Archive old database data on nova-cloud-controller
+        OVN subordinate upgrade plan
+            Upgrade plan for 'ovn-chassis' to 'victoria'
+                Refresh 'ovn-chassis' to the latest revision of '22.03/stable'
         Control Plane subordinate(s) upgrade plan
             Upgrade plan for 'keystone-ldap' to 'victoria'
                 Refresh 'keystone-ldap' to the latest revision of 'ussuri/stable'
```

Comment:
Why not simply move the OVN subordinate applications into the control plane subordinates list?  This would solve the problem without any added complexity.

Reply:
Because they are not control plane subordinate as I understand, if we follow the definition of control/data plane and principle/subordinate.

Reply:
If that's the only reason, then I vote we put it in the control plane subordinates list with a comment about why it's in that list.

Reply:
Two more reasons:

- I am thinking the OVN(ovn-central, ovn-chassis, ...) upgrade can be a separate, independent upgrade. We may lose some flexibility if we want to do this in the future.
- I don't like to make upgrade group definition become more vague. To me it's more clear not the group them together.

Reply:
> I am thinking the OVN(ovn-central, ovn-chassis, ...) upgrade can be a separate, independent upgrade.

What are the use cases for this?  I don't think we have a case where you only want to upgrade ovn.

> I don't like to make upgrade group definition become more vague. 

Our upgrade groups are already arbitrary.  I think the main reason is so we can upgrade the hypervisors individually.  Outside of that, we don't have to follow any other constraint.

I don't want to add all this extra complexity just to change the upgrade order.

Reply:
> What are the use cases for this? I don't think we have a case where you only want to upgrade ovn.

If we want to support ovn 20.04 -> 22.03

> I don't want to add all this extra complexity just to change the upgrade order.

Putting ovn-chassis in `apps_control_plane` also increase the complexity in code. I don't see big issue to add one more upgrade group before control plane subordinate, to me it's equal on the upgrade order and avoid the misunderstand of the control/data plane and principle/subordinate.

Reply:
Ok I see how the current design of COU makes it different to cleanly implement this.